packages/store/src/lib/store.ts

```diff
@@ -36,6 +37,10 @@ export class Store<
     const nextState = reducers.reduce((value, reducer) => {
       value = reducer(value, this.context);
 
+      if (elfHooksRegistry.preStoreUpdate) {
```

Comment:
Let's do it outside the reducer and pass the prev and new state:

  ` nextState = elfHooksRegistry.preStoreUpdate(currenState, nextState, { name: this.name });`